### PR TITLE
Fixes API misuse and logic errors

### PR DESCRIPTION
**Pull request by akallabeth**
* Fixes allocations of zero size
* Fixes uninitialized memory errors
* Fixes ```_comm_ioctl_tcsetattr```, compared ```struct termios``` against ```struct termios*```

### CODE REVIEW

File: winpr/libwinpr/comm/comm_ioctl.c [hunk 1]
```diff
@@ -709,7 +709,7 @@ int _comm_ioctl_tcsetattr(int fd, int optional_actions, const struct termios *te
 		return result;
 	}
 
-	if (memcmp(&currentState, &termios_p, sizeof(struct termios)) != 0)
+	if (memcmp(&currentState, termios_p, sizeof(struct termios)) != 0)
```
Comment (bmiklautz) [inline winpr/libwinpr/comm/comm_ioctl.c]: Good catch!

**Comment by bmiklautz** [comment]
+1

**Comment by bmiklautz** [comment]
@freerdp-bot test

**Comment by MartinHaimberger** [comment]
+1 LGTM

**Pull request merged**